vertical-pod-autoscaler/enhancements/XXXX-restore-selector-support.md

```diff
@@ -0,0 +1,105 @@
+# AEP-XXXX: Restore Label Selector Support to VPA
+
+## Summary
+
+This proposal enhances the VerticalPodAutoscalerSpec to allow the co-existence of TargetRef and Selector (present in v1beta1 but removed in v1). When both are present, the Selector acts as a filter applied to the Pods managed by the TargetRef. This allows multiple VPA objects to manage disjoint subsets of a single Controller's pods (e.g., partitioning a StatefulSet/Deployment into Leaders and Followers).
+
+## Motivation
+
+Currently, VPA relies exclusively on `targetRef` to identify Pods. This enforces a 1:1 relationship between the VPA and a Workload Controller (Deployment, StatefulSet, etc.).
+
+While this is sufficient for stateless workloads, it fails for **Heterogeneous Stateful Workloads** where pods in the same controller perform different roles with different resource footprints.
```

Comment:
```suggestion
While this is sufficient for stateless workloads, it fails for **Heterogeneous Workloads** where pods in the same controller perform different roles with different resource footprints.
```

I think it may make sense to remove the reference to "stateful" as this proposal should also address vertical autoscaling for stateless workloads that have heterogeneous resource requirements under the same controller.